### PR TITLE
ENG-1270: adding user preferences and connecting to API

### PR DESCRIPTION
**Pull request by gReis89**

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **greis89**

If they are not already assigned, you can assign the PR to them by writing `/assign @greis89` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/ui/app/App.js
Comment (ichalagashvili) [inline src/ui/app/App.js:347]: I remember this was purposely changed to `componentDidUpdate` because there is some kind of error when fetching the `plugins` because of some auth or smth. Can you revert it back and write preferences fetching in `componentDidMount` ? Important is, that `fetchPlugins` is inside `componentDidUpdate`

**Review by ichalagashvili — CHANGES_REQUESTED** [review]

File: src/ui/app/App.js
Comment (gReis89) [reply src/ui/app/App.js:347]: I checked with @gidesan, and we said it would be ok to move to `componentDidMount`. He might not be aware of that though, will keep the `fetchPlugins` on componentDidUpdate to avoid any issues.

**Review by ichalagashvili — APPROVED** [review]
looks good!

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/app-builder/pull/873#pullrequestreview-512059200" title="Approved">ichalagashvili</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)~~ [ichalagashvili]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
